presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbSourceExactMatchSelector.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.db;
+
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupSelector;
+import com.facebook.presto.spi.resourceGroups.SelectionContext;
+import io.airlift.json.JsonCodec;
+import io.airlift.log.Logger;
+import org.skife.jdbi.v2.exceptions.DBIException;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.airlift.units.Duration.nanosSince;
+import static java.util.Objects.requireNonNull;
+
+public class DbSourceExactMatchSelector
+        implements ResourceGroupSelector
+{
+    private static final Logger log = Logger.get(DbSourceExactMatchSelector.class);
+    private static final JsonCodec<ResourceGroupId> resourceGroupIdCodec = JsonCodec.jsonCodec(ResourceGroupId.class);
+    private final ResourceGroupsDao dao;
+    private final String environment;
+    private final AtomicReference<Long> daoOfflineStart = new AtomicReference<>();
```

Comment:
why don't you use `AtomicLong`?

Reply:
Code in this class is copy-pasted. I didn't read, I don't remember...

Reply:
Seems like this value can be null, But AtomicLong can't.

Reply:
you can still init with -1 or some other sentinel value (long min value) instead of null.